added random playlist example in json-api.md

AlexeyMal opened the pull request:
added random playlist example

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Added a JSON API example illustrating an endless playlist that applies a preset indefinitely, with randomized effects and color/palette selection, detailed segment and brightness configuration, and step-by-step instructions for uploading the payload and applying the playlist via the API.

<!-- review_stack_entry_start -->

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/wled/WLED-Docs/pull/276)

<!-- review_stack_entry_end -->
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

AlexeyMal committed:
Update json-api.md

added random playlist example

softhack007 committed:
Merge branch 'main' into pr/276

AlexeyMal commented on docs/interfaces/json-api.md:
````diff
@@ -345,6 +345,17 @@ transition | Array of time each preset should transition to the next one, in ten
 repeat | How many times the entire playlist should cycle before finishing. Set to `0` for an indefinite cycle. Default to indefinite if not provided.
 end | Single preset ID to apply after the playlist finished. Has no effect when an indefinite cycle is set. If not provided, the light will stay on the last preset of the playlist.
 
+Sample playlist for an endless cycle of random effects with random colors and random palettes:
+
+```json
+{
+"9":{"playlist":{"ps":[12],"dur":[600],"transition":[30],"repeat":0,"r":false,"end":0},"on":true,"n":"AllRandomPlaylist"},
+
+"12":{"bri":255,"mainseg":0,"on":true,"seg":[{"bri":255,"c1":128,"c2":128,"c3":16,"cct":127,"col":["r",[0,0,0], "r"],"frz":false,"fx":"r","grp":1,"id":0,"ix":128,"m12":1,"mi":false,"o1":false,"o2":false,"o3":false,"of":0,"on":true,"pal":"r","rev":false,"sel":true,"set":0,"si":0,"spc":0,"start":0,"stop":50,"sx":128 ,"fxdef":true}],"transition":7,"n":"random effect"}
+}
+```
````
The sample playlist JSON includes an undocumented "r" field inside the playlist object. Remove that "r" property from the example.

AlexeyMal replied:
Verify each finding against the current code and only fix it if needed.

In `@docs/interfaces/json-api.md` around lines 348 - 356, The example uses
top-level numeric keys "9" (playlist object with "playlist":{...}) and "12"
(preset/preset segment under "seg":{...}); add one short clarifying sentence
below the JSON explaining that those numeric keys are resource IDs (playlist ID
9 and preset ID 12), that POSTing this payload to /json/cfg will persist/save
both the playlist and preset, and that to apply a playlist immediately you would
instead POST an appropriate state update to /json/state (or use the "playlist"
command) — reference the "9" playlist and "12" preset in the sentence so readers
know which parts are persisted vs applied.

AlexeyMal replied:
apply your changes to this pull request

softhack007 replied:
@AlexeyMal you might have found out already - coderabbit is a review expert AI, but it won't rewrite the PR for you. This is intended :-)

softhack007 replied:
> Please add one short sentence under the code block (or adjust the example heading) clarifying the intended usage and endpoint so users don’t copy a payload that does nothing.

@coderabbitAI it is not necessary to upload a (single) custom preset/playlist (JSON format) via network API, /edit or /json/ endpoints. The simpler solution is to directly enter JSON into the WebUI when saving a preset (un-check "Use current state" => textbox "API command"). see my comment https://github.com/wled/WLED-Docs/pull/276#discussion_r3229660176.

AlexeyMal committed:
Clarify JSON payload example for playlists and presets

Updated JSON example in documentation to clarify resource IDs and usage instructions.

AlexeyMal reviewed (commented):
resolved coderabbitai reported issues

softhack007 commented on docs/interfaces/json-api.md:
````diff
@@ -349,12 +349,12 @@ Sample playlist for an endless cycle of random effects with random colors and ra
 }
 ```
-
+These numeric keys (`"9"` and `"12"`) are resource IDs — playlist ID 9 and preset ID 12 — stored in WLED's internal `presets.json` file; upload this payload to your device via the `/edit` page to persist both entries, then apply the saved playlist by POSTing `{"ps":9}` to `/json/state`. Alternatively, use the presets page web GUI to enter json code and to activate the playlist.
````
You can do it simpler: use the UI to save a new preset, uncheck "Use current state" => textbox "API command" appears. Now copy/paste JSON into the API command textarea, then press "save".

<img width="282" height="483" alt="Image" src="https://github.com/user-attachments/assets/c101c901-59c7-46c2-b602-21f8e4a192bd" />

AlexeyMal committed:
Revise preset saving and activation instructions

Updated instructions for saving presets and activating playlists in the JSON API documentation.
